typed/typed.go

```diff
@@ -76,36 +44,38 @@ func AsTyped(v value.Value, s *schema.Schema, typeRef schema.TypeRef) (TypedValu
 // conforms to the schema, for cases where that has already been checked or
 // where you're going to call a method that validates as a side-effect (like
 // ToFieldSet).
-func AsTypedUnvalidated(v value.Value, s *schema.Schema, typeRef schema.TypeRef) TypedValue {
-	tv := typedValue{
+func AsTypedUnvalidated(v value.Value, s *schema.Schema, typeRef schema.TypeRef) *TypedValue {
+	tv := &TypedValue{
 		value:   v,
 		typeRef: typeRef,
 		schema:  s,
 	}
 	return tv
 }
 
-// typedValue is a value of some specific type.
-type typedValue struct {
+// TypedValue is a value of some specific type.
+type TypedValue struct {
 	value   value.Value
 	typeRef schema.TypeRef
 	schema  *schema.Schema
 }
 
-var _ TypedValue = typedValue{}
-
-func (tv typedValue) AsValue() *value.Value {
+// AsValue removes the type from the TypedValue and only keeps the value.
+func (tv *TypedValue) AsValue() *value.Value {
```

Comment:
I don't think it matters, but this is now always returning the same value (while it wasn't before, because it was returning a pointer in the copy)?